language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -386,11 +388,20 @@ genDefDataType curPkgId conName mod tpls def =
                             ] ++
                             ["};"]
                         associatedTypes =
-                          maybe [] (\key ->
-                              [ "export namespace " <> conName <> " {"
-                              , "  export type Key = " <> fst (genType (moduleName mod) (tplKeyType key)) <> ""
+                          case tplKey tpl of
+                            Nothing -> []
```

Comment:
I don’t understand this, why do we not want these synonyms if there is no contract key? Surely I can still get events for a template without contract keys?

Reply:
No you are absolutely right. Dumb mistake. Will fix.